### DISCUSSION TITLE
Invoke a function variable

### DISCUSSION THREAD

**Discussion by oscarotero** [harttle/liquidjs#580] [Q&A]
Hi. In a Nunjucks template I have some variables that are functions, that I can invoke to get the final value. For example:

```
{% set pages = getPages() %}
```

The same code in Liquid throws an error:
```
{% assign pages = getPages() %}

-> unexpected token ")"
```

I guess this is not supported in liquidjs. Is there any way to run functions in a template. I think it's a very useful feature.

**Comment by harttle** [comment]
Hi @oscarotero , we don't use `()` to call functions in Liquid, the same as in Ruby. Here's a test case:

```typescript
    const liquid = new Liquid()
    const tpl = `{% assign foo = bar %}foo: {{ foo }}`
    const ctx = { bar: () => 'BAR' }
    const html = await liquid.parseAndRender(tpl, ctx)
    expect(html.trim()).to.equal('foo: BAR')
```

We added this feature in #222 , it should be available since a very old version. Try it!

**Reply by oscarotero** [reply]
Ah, thanks @harttle for your reply. I didn't know this feature (too magic to my taste but ok:)

In my real use case, I need to pass arguments to the function. Is that possible?

**Reply by harttle** [reply]
Not out of the box. Liquid language is not in imperative style like calling a function with `func()`. But there's workaround as mentioned in https://groups.google.com/g/liquid-templates/c/kK5oTnVLbeU/m/FyJuE25NYqAJ

And there's a mechanism named `Drop` (not documented much on liquidjs.com but documented on [Shopify Liquid](https://github.com/Shopify/liquid/wiki/Introduction-to-Drops)), there's a [liquidMethodMissing](https://liquidjs.com/zh-cn/api/classes/drop_drop_.drop.html#liquidMethodMissing) method which can be used to substitute method parameter to some extent.

You can find more info in this PR: https://github.com/harttle/liquidjs/pull/285

**Reply by oscarotero** [reply]
Thanks for all these references. I'm not sure if that can help because seems that `Drop` only accept one argument?

To give more context, I want to be able to use this [search object](https://lume.land/plugins/search/#limit-the-results) inside a liquid template. It's a class with several methods and some of them have 2 or three arguments (query, sort and limit). For example:

```njk
{% for page in search.pages("category=javascript", "order title", 10) %}
  ...
{% endfor %}
```

I'm the maintainer of Lume (a static site generator) and it has a plugin to use Liquidjs as the template engine, but seems like it's not possible to use the helpers like `search`.
